package/cloudshell/cm/ansible/domain/playbook_downloader.py

```diff
@@ -67,10 +67,11 @@ def _download(self, url, auth, logger, cancel_sampler, verify_certificate):
         if not response_valid and auth is None:
             raise Exception('Please make sure the URL is valid, and the credentials are correct and necessary.')
 
+        generic_auth = auth.token if auth.token else auth.password
```

Comment:
there is a different section intended to handle auth password further down, if the normal token path does not succeed.
What is the intention of using this path for auth password as well?

Reply:
for supporting older customers that can't upgrade yet to also use tokens